packages/now-cli/src/util/dev/server.ts

```diff
@@ -545,18 +526,13 @@ export default class DevServer {
 
     // no builds -> zero config
     if (!config.builds || config.builds.length === 0) {
-      const detectorResult = await detectDefaults({
-        fs: new DevDetectorFilesystem(this.cwd, files),
-      });
+      const { projectSettings } = config;
 
-      const { builders, warnings, errors } = await detectBuilders(
-        files,
-        detectorResult,
-        {
-          tag: getDistTag(cliVersion) === 'canary' ? 'canary' : 'latest',
-          functions: config.functions,
-        }
-      );
+      const { builders, warnings, errors } = await detectBuilders(files, pkg, {
+        tag: getDistTag(cliVersion) === 'canary' ? 'canary' : 'latest',
+        functions: config.functions,
+        ...(projectSettings ? { projectSettings } : {}),
```

Comment:
This is an interesting technique. Are you doing this to avoid setting `projectSettings : undefined`?

Reply:
yes